### PR TITLE
Create dev folder, move dev scripts and docs there

### PR DESCRIPTION
**Pull request by Amorymeltzer**
Allow `sync.pl` and `patch-test.js` to be run from any subdirectory, and move them into a new dev folder, along with morebits-test and DEVELOPER.md.  oldtwinkle.js has been removed as entirely unnecessary.

Also allows providing a port number to patchtest via commandline, and adds sync.pl to the package defs as `sync` and `deployall` with the `--mode=deploy --all` options (use `npm run deployall -- --lang=`, etc. to pass other flags).

Instruct `eslint` to ignore /dev folder

----

I've been sitting on this for a while, given the additional scripts in #911, but since @MusikAnimal [mentioned](https://github.com/azatoth/twinkle/issues/129#issuecomment-703293623) the testing method not being noted, I figured I'd toss it up.  It worked over the summer!

### CODE REVIEW

**Comment by siddharthvp** [comment]
I would suggest using the standard conventions of an `src`, `lib` (for select2), `tests`, `scripts` (for the dev scripts) and `.github` (for contributing.md, security.md, etc) directories, rather than a `dev`. 

Also, `Gadget.md` should probably be called `gadget-definition.txt` (the content isn't markdown).

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -33,11 +33,11 @@ Once you've got a local fork up and running, commit your changes!
 
 ### Testing your code
 
-Testing Twinkle can be tricky, but the most straightforward way to test your code is to open up your [browser's console window][jserrors] and paste in your new code.  You'll have to load the new version by running the corresponding function in your console, e.g., `Twinkle.protect()` for twinkleprotect.js.
+Testing Twinkle can be tricky, but thankfully we've made it easy with a helper script, [patch-test.js](./dev/patch-test.js).  Simply run `npm run patchtest` in your terminal, which will generate the `patch-test-loader.js` file.  Then, run a localhost server (e.g. `php -S 127.0.0.1:5500` or `python -m SimpleHTTPServer 5500`) and paste the contents of `patch-test-loader.js` in your [browser's console window][jserrors].  You should be all set!  You can also test your code by simply pasting it into the browser console, although that's not recommended.
```
Comment (siddharthvp) [inline CONTRIBUTING.md:36]: This system will likely change when we have a transpilation step, but while this is here, it should mention that they also need to put the importScript statement in their common.js page for the locally hosted to actually load.

**Comment by Amorymeltzer** [comment]
>tests, scripts (for the dev scripts) and .github (for contributing.md, security.md, etc) directories, rather than a dev.

YMMV, but I think having a "hey devs here's some shit for you" folder is more convenient rather than splitting everyone up when we've only got a handful of items; the main goal is not to pollute the base directory.  That being said, I didn't know GitHub also detects contributing/security/etc. in the .github directory, so that's neat, although I do think have CONTRIBUTING.md front-and-center is slightly more helpful for discovery.  You're suggesting:

- .github/CONTRIBUTING.md
- .github/SECURITY.md
- .github/DEVELOPER.md
- scripts/sync.pl
- tests/patch-test.js
- tests/morebits-test.js

Yeah?

>an `src`, `lib` (for select2)

Something got garbled here, but yeah; moved to `lib`.

>Also, Gadget.md should probably be called gadget-definition.txt (the content isn't markdown).

Indeed.  Done.

File: CONTRIBUTING.md (hunk 1 above)
Comment (Amorymeltzer) [reply CONTRIBUTING.md:36]: You mean `mw.loader.getScript('http://127.0.0.1:5500/dev/patch-test-loader.js');`, right?

```suggestion
Testing Twinkle can be tricky, but thankfully we've made it easy with a helper script, [patch-test.js](./dev/patch-test.js).  Simply run `npm run patchtest` in your terminal, which will generate the `patch-test-loader.js` file.  Then, run a localhost server (e.g. `php -S 127.0.0.1:5500` or `python -m SimpleHTTPServer 5500`) and paste the contents of `patch-test-loader.js` in your [browser's console window][jserrors].  You should be all set!  If you put `mw.loader.getScript('http://127.0.0.1:5500/dev/patch-test-loader.js');` in your common.js page, you won't even need to paste anything once you run the server!  You can also test your code by simply pasting it into the browser console, although that's not recommended.
```

Comment (siddharthvp) [reply CONTRIBUTING.md:36]: I did miss the "paste the contents of `patch-test-loader.js` in your [browser's console window][jserrors]" part, so okay, looks good now!

**Comment by siddharthvp** [comment]
`scripts/patch-test.js` (not tests/) otherwise yeah, that's what i was suggesting, but the present setup looks okay too. Though to be politically correct, there should be a select2 folder inside the lib folder, and maybe rename `select2-LICENSE.md` to the standard `LICENSE`, and add a `PATCHES` describing the changes made to the original.

**Comment by Amorymeltzer** [comment]
>the present setup looks okay too

You mean the `dev` or the examples in [my comment](https://github.com/azatoth/twinkle/pull/1142#issuecomment-706718740)?

**Comment by siddharthvp** [comment]
>

**Comment by Amorymeltzer** [comment]
@siddharthvp ?

**Comment by siddharthvp** [comment]
Oh, not sure how that one above went empty. But yeah, I meant the `dev`. I went ahead and merged this (before the conversation can go more awkward!). More changes like adding an `src` and `tests` can come later I guess.

**Pull request merged**